src/emcee.jl

```diff
@@ -0,0 +1,227 @@
+struct Ensemble{D} <: MHSampler
+    n_walkers::Int
+    proposal::D
+end
+
+struct Walker{T<:Union{Vector,Real,NamedTuple},L<:Real}
+    params::T
+    lp::L
+end
+
+Walker(t::Transition) = Walker(t.params, t.lp)
+Walker(model::DensityModel, params) = Walker(params, logdensity(model, params))
+
+# Define the first step! function, which is called at the 
+# beginning of sampling. Return the initial parameter used
+# to define the sampler.
+function AbstractMCMC.step!(
+    rng::Random.AbstractRNG,
+    model::DensityModel,
+    spl::Ensemble,
+    N::Integer,
+    ::Nothing;
+    init_params = nothing,
+    kwargs...,
+)
+    if init_params === nothing
+        return propose(spl, model)
```

Comment:
Better use the provided RNG (as in most/all other parts of AdvancedMH so far):
```suggestion
        return propose(rng, spl, model)
```